build.ps1

```diff
@@ -65,9 +65,10 @@ Param(
     [string] $verbosity = "q"
 )
 
-Function IsWindowsDevelopmentBox()
+Function IsWindowsDesktopDevelopmentBox()
 {
-    return ([Environment]::OSVersion.Platform -eq [System.PlatformID]::Win32NT)
+    $osEdition = (Get-ItemProperty 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion').EditionID
```

Comment:
Could you please explain this section? And also add a comment (for future reference).

Reply:
Here we use reg key HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\EditionID to tell the different Windows editions apart - _"IoTUAP"_ is Windows IoT Core. Windows Desktop would have this reg key set to _"Enterprise"_, or _"Professional"_ or etc.